internal/utils/exec.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+)
+
+func Exec(log func(format string, a ...interface{}), command string, args ...string) ([]byte, error) {
```

Comment:
Perhaps this could be implemented on `K8sInstaller` instead? This is only ever called as `utils.Exec(k.Log, ...)`, so why not make make that `k.Exec(...)` with some logging code instead? Then this can become `utils.Exec(command string, args ...string) ([]byte, error)`

Reply:
It's also called from `Parameter` methods in the `hubble` package. I guess we could duplicate the logic in `Parameter` as well, but `utils.Exec(k.Log, ...)` seems nicely concise and gets rid of the boilerplate.

We could probably introduce a `Logger` interface which both `K8sInstaller` and `hubble.Parameter` implement and then use that in `utils.Exec`, i.e. `func Exec(l Logger, command string, args ...string) ([]byte, error)` and then use `l.Log` inside. That might make the intention a bit clearer? WDYT?

Reply:
Sounds good! Why not add a `K8sInstaller.Exec()` wrapper that avoids the repetitive `k.Log` as the first parameter?

Reply:
Good point, added in the latest revision.